packages/code-du-travail-frontend/src/outils/common/situations.utils.js

```diff
@@ -112,3 +112,23 @@ export function getRef({ ref, refUrl }) {
     </p>
   );
 }
+
+export const resetFieldForm = ({
+  form,
+  key,
+  initialSituations,
+  pastQuestions
+}) => {
+  form.batch(() => {
+    // list keys that no longer exist
+    const resetFormProps = Object.keys(initialSituations)
+      .filter(k => !pastQuestions.find(([key]) => k === key))
+      .concat(
+        // list keys that need to be reseted
+        pastQuestions
+          .slice(pastQuestions.findIndex(([k]) => k === key) + 1)
+          .map(([key]) => key)
+      );
+    resetFormProps.forEach(key => form.change(`criteria.${key}`, undefined));
```

Comment:
`criteria` should be extracted unless we want to forget to update it is forms data change ;)

Reply:
Of course